integration-tests/hibernate-tenancy/src/main/java/io/quarkus/it/hibernate/multitenancy/inventory/InventoryTenantResolver.java

```diff
@@ -0,0 +1,41 @@
+package io.quarkus.it.hibernate.multitenancy.inventory;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Unremovable;
+import io.quarkus.hibernate.orm.PersistenceUnit;
+import io.quarkus.hibernate.orm.runtime.tenant.TenantResolver;
+import io.vertx.ext.web.RoutingContext;
+
+@RequestScoped
+@Unremovable
```

Comment:
Is this needed? If so, then we probably want to mention it in the example in the docs as well

Reply:
So, it used to be like that and I didn't change it. I added a commit to make the beans automatically unremovable. Could you have a look at this commit?

Reply:
Seems perfectly reasonable!